tests/test_docker.py

```diff
@@ -180,6 +180,15 @@ def test_build_with_buildargs(self):
         client = docker.from_env(
             version='auto',
             timeout=180)
-        self.assertEqual(
-            'somevalue',
-            client.containers.run(im, remove=True).decode())
+
+        # To capture all output to inspect, must delay removal until
+        # after retrieval of logs otherwise the API can sometimes return
+        # an empty result
+        c = client.containers.create(im, auto_remove=False, detach=False)
```

Comment:
Should this be not detach=True ?? https://docker-py.readthedocs.io/en/stable/containers.html

Reply:
Wait it to wait until the container is finished, that way the `c.start()` below becomes blocking, otherwise need to call `c.wait()` in case the logs are retrieved before output is produced.

Reply:
Only if you use `run`, note that I've using `create` here which will always return a container object, see https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.ContainerCollection.create

Reply:
I missed that